aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -538,19 +597,74 @@ def clear_context(self):
         self.context.clear()
 
 
-class MiddlewareHandler:
-    """Represents a callable Middleware function used to build a middleware stack when a route is resolved"""
+class MiddlewareStackWrapper:
+    """
+    creates a Middle Stack Wrapper instance to be used as a "Frame" in the overall stack of
+    middleware functions.  Each instance contains the current middleware to be executed and the next
+    middleware function to be called in the stack.
+
+    In this way the middleware stack is constructed in a recursive fashion, with each middleware
+    calling the next as a simple function call.  The actual Python call-stack will contain
+    each MiddlewareStackWrapper "Frame", meaning any Middleware function can cause the
+    entire Middleware call chain to be exited early (short-circuited) by raising an exception
+    or by simply returning early with a custom Response.  The decision to short-circuit the middleware
+    chain is at the user's discretion but instantly available due to the Wrapped nature of the
+    callable constructs in the Middleware stack and each Middleware function having complete control over
+    whether the "Next" handler in the stack is called or not.
+
+    Parameters
+    ----------
+    handler : Callable
+        The API route handler to be invoked when this class is executed.
+    next_middleware : Callable
+        The next middleware in the middleware stack.
+    """
 
     def __init__(
         self,
-        handler: Callable[..., Any],
-        next_handler: Callable[..., Any],
+        handler: Callable,
```

Comment:
```suggestion
        current_middleware: Callable,
```

Makes it easier when debugging